modules/vector-sets/hnsw.h

```diff
@@ -17,10 +17,10 @@
 #include <pthread.h>
 #include <stdatomic.h>
 
-#define HNSW_DEFAULT_M  16     /* Used when 0 is given at creation time. */
-#define HNSW_MIN_M      4      /* Probably even too low already. */
-#define HNSW_MAX_M      4096   /* Safeguard sanity limit. */
-#define HNSW_MAX_THREADS 32    /* Maximum number of concurrent threads */
+#define HNSW_DEFAULT_M   16                             /* Used when 0 is given at creation time. */
+#define HNSW_MIN_M       4                              /* Probably even too low already. */
+#define HNSW_MAX_M       4096                           /* Safeguard sanity limit. */
+#define HNSW_MAX_THREADS 32                             /* Maximum number of concurrent threads, including main thread */
```

Comment:
There is no notion of "main thread" in this implementation. Every thread is the same.